### PR TITLE
UX/feedback- Custom datepicker added for manual entry

### PR DESCRIPTION
**Pull request by Shruti-Apte**
What :
Date picker updated according to designs for manual entry.

Why :
It was not matching the designs.
UX was not correct. (We were only able to open the datepicker when clicking on the calendar icon and not the whole field).

Preview:

Screenshot 2023-02-09 at 11 19 23 AM
https://www.loom.com/share/7418c19ad02046d385ff358e242142e7 

TODOs : 
- Datepicker functionality 
- Outside click

### CODE REVIEW

File: app/javascript/src/components/Invoices/common/ManualEntry/index.tsx [hunk 1]
```diff
@@ -28,9 +31,11 @@ const ManualEntry = ({
   const [lineTotal, setLineTotal] = useState<string>("0");
   const [qtyInHHrMin, setQtyInHHrMin] = useState<any>(minToHHMM(quantity));
   const [isEnter, setIsEnter] = useState<boolean>(false);
-  const ref = useRef(null);
+  const [showDatePicker, setShowDatePicker] = useState<boolean>(false);
   const wrapperRef = useRef(null);
 
+  const formattedDate = date => dayjs(date).format("MM.DD.YYYY");
```
Comment (vipulnsward) [inline app/javascript/src/components/Invoices/common/ManualEntry/index.tsx:37]: These all formats need to come via a global config. I will work on migrating them.

**Comment by vipulnsward** [comment]
@Shruti-Apte please rebase

File: app/javascript/src/components/Invoices/common/ManualEntry/index.tsx (hunk 1 above)
Comment (Shruti-Apte) [reply app/javascript/src/components/Invoices/common/ManualEntry/index.tsx:37]: What should I do for now? on some pages date format is coming through API.

**Comment by supriya3105** [comment]
@Shruti-Apte Please check if the date format is coming from backend.

File: app/javascript/src/components/Invoices/common/ManualEntry/index.tsx (hunk 1 above)
Comment (Shruti-Apte) [reply app/javascript/src/components/Invoices/common/ManualEntry/index.tsx]: On edit invoice and on generate invoice page we are using this component. One of the API has date_format updated the code!

**Review by vipulnsward — APPROVED** [review]

**Pull request merged**